tests/test_validators.py

```diff
@@ -1155,3 +1155,38 @@ class Model(BaseModel):
             'ctx': {'given': 'nope', 'permitted': ('foo', 'bar')},
         }
     ]
+
+
+def test_field_that_is_being_validated_is_excluded_from_validator_values():
+    validators_values = None
+
+    class Model(BaseModel):
+        foo: str
+        bar: str
+
+        @validator('foo')
+        def validate_foo(cls, v, values):
+            nonlocal validators_values
+            validators_values = values
+
+    Model(foo='foo_value', bar='bar_value')
+    assert validators_values == {'bar': 'bar_value'}
```

Comment:
There are two issues here:
- you don't test with `validate_assignement = True`
- you use the reference of `values` so it won't work as `values` is then updated before being checked with `assert`

Btw you could simply use a mock to check the validation. Something like
```python
def test_field_that_is_being_validated_is_excluded_from_validator_values(mocker):
    check_values = mocker.MagicMock()

    class Model(BaseModel):
        foo: str
        bar: str

        class Config:
            validate_assignment = True

        @validator('bar')
        def validate_bar(cls, v, values):
            check_values({**values})
            return v

    model = Model(foo='foo_value', bar='bar_value')
    check_values.reset_mock()

    model.bar = 'new_bar_value'
    check_values.assert_called_once_with({'foo': 'foo_value'})
```

Reply:
Apologies, I didn't have my local venv set up, so these changes were untested.

Just pushed up a new set of changes now, and I used a mock to capture the calls like you suggested.

Reply:
No worries. Thanks a lot for this!